src/transformers/models/llava/modeling_llava.py

```diff
@@ -344,6 +344,12 @@ def _merge_input_ids_with_image_features(self, image_features, inputs_embeds, in
         final_attention_mask |= image_to_overwrite
         position_ids = (final_attention_mask.cumsum(-1) - 1).masked_fill_((final_attention_mask == 0), 1)
 
+        # 6. Mask out the embedding at padding positions, as we later use the past_key_value value to determine the non-attended tokens.
```

Comment:
A bit later in the code (specifically here: https://github.com/huggingface/transformers/blob/aa17cf986f0761382f3d6e591e985a42671c3fb7/src/transformers/models/llava/modeling_llava.py#L449), we check `past_key_values` value to find unattended tokens. This was not correct before because this added step `6.` was missing.

Overall, this would be worth a larger refactor that would avoid regenerating full masks at every forward step in the generate. This PR is just a hotfix.